modules/statistics/php/NDB_Form_statistics.class.inc

```diff
@@ -545,19 +548,48 @@ class NDB_Form_Statistics extends NDB_Form
         } else {
             $Param_Project = '';
         }
-        $subprojects = Utility::getSubprojectsForProject($currentProject);
+        $subprojects[null] = 'All Valid Sub-Projects';
+
+        foreach (Utility::getSubprojectsForProject($currentProject)
+            as $key=>$value) {
+            $subprojects[$key] = $value;
+        }
 
-        $this->tpl_data['Subprojects'] = $subprojects;
         foreach ($subprojects as $key=>$val) {
             $subprojList .= $key.",";
         }
-        $subprojList = substr($subprojList, 0, -1);
-        if (!empty($subprojList)) {
-            $suproject_query ="AND s.SubprojectID IN ($subprojList)";
+
+        $this->tpl_data['Subprojects'] = $subprojects;
+        $currentSubProject = null;
+        $subprojList       = substr($subprojList, 1, -1);
+        if (isset($_REQUEST['BehaviouralSubproject'])
+            && $_REQUEST['BehaviouralSubproject'] != ''
```

Comment:
`!empty($var['x'])` doesn't have the same truth value as `isset($var['x']) && $var['x'] != '')`. It has the same truth value as `$var != ''`, but suppresses warnings if the index x isn't set. See https://www.virendrachandak.com/techtalk/php-isset-vs-empty-vs-is_null/ for a table of the truth values of the (somewhat ridiculous) number of ways PHP has to check if a variable's value is empty.

`$REQUEST['BehaviouralSubproject'] ?? ''`, on the other hand, does have the same truth value as `isset(x) && x`. It's how the `??` operator is defined in PHP (since PHP7..)